messaging/src/main/java/org/axonframework/eventhandling/DefaultEventProcessorSpanFactory.java

```diff
@@ -0,0 +1,199 @@
+
+    import org.axonframework.common.BuilderUtils;
+    import org.axonframework.tracing.NoOpSpanFactory;
+    import org.axonframework.tracing.Span;
+    import org.axonframework.tracing.SpanFactory;
+
+    import java.time.Duration;
+    import java.time.Instant;
+    import java.util.Comparator;
+    import java.util.List;
+    import java.util.function.Function;
+    import java.util.function.Supplier;
+
+    /**
+     * Default implementation of the {@link EventProcessorSpanFactory}.
+     * <p>
+     * Creates a root trace for each batch of events and a linked handler span for each event in the batch by default. This
+     * behavior can be changed by setting {@link Builder#disableBatchTrace(boolean)} to {@code true}.
+     * <p>
+     * If you prefer to have asynchronously handled events in the same trace as the trace that published it, you can set
+     * {@link Builder#distributedInSameTrace(boolean)} to {@code true}. If {@code true}, this duration defaults to 2
+     * minutes, which you can adjust using {@link Builder#distributedInSameTraceTimeLimit(Duration)}.
+     *
+     * @author Mitchell Herrijgers
+     * @since 4.9.0
+     */
+    public class DefaultEventProcessorSpanFactory implements EventProcessorSpanFactory {
+
+        private final SpanFactory spanFactory;
+        private final boolean disableBatchTrace;
+        private final boolean distributedInSameTrace;
+        private final Duration distributedInSameTraceTimeLimit;
+
+        /**
+         * Creates a new {@link DefaultEventProcessorSpanFactory} using the provided {@code builder}.
+         *
+         * @param builder The builder to build the {@link DefaultEventProcessorSpanFactory} from.
+         */
+        protected DefaultEventProcessorSpanFactory(Builder builder) {
+            builder.validate();
+            this.spanFactory = builder.builderSpanFactory;
+            this.disableBatchTrace = builder.builderDisableBatchTrace;
+            this.distributedInSameTrace = builder.builderDistributedInSameTrace;
+            this.distributedInSameTraceTimeLimit = builder.builderDistributedInSameTraceTimeLimit;
+        }
+
+
+        /**
+         * Creates a new {@link Builder} to build a {@link DefaultEventProcessorSpanFactory} with. The default values are:
+         * <ul>
+         *     <li>{@code disableBatchTrace} defaults to {@code false}</li>
+         *     <li>{@code distributedInSameTrace} defaults to {@code false}</li>
+         *     <li>{@code distributedInSameTraceTimeLimit} defaults to 2 minutes</li>
+         * </ul>
+         * The {@code spanFactory} is a required field and should be provided.
+         *
+         * @return The {@link Builder} to build a {@link DefaultEventProcessorSpanFactory} with.
+         */
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        @Override
+        public Span createBatchSpan(boolean streaming, List<? extends EventMessage<?>> eventMessages) {
```

Comment:
We didn't do _any_ batch span before this PR, right?

Reply:
Correct, this is new

Reply:
Great addition I think!